### PR TITLE
Add two private attributes for SR-ZG9040A/ZG9041A-D.

### PR DESCRIPTION
**Pull request by maginawin**
1. External Switch Type
2. Minimum PWM

### CODE REVIEW

File: src/devices/sunricher.ts [hunk 1]
```diff
@@ -25,13 +25,150 @@ import {
...
+            .withLabel('Minimum PWM')
+            .withDescription('Power off the device and wait for 3 seconds before reconnecting to apply the settings.')
+            .withValueMin(0)
+            .withValueMax(50)
+            .withUnit('%')
+            .withValueStep(1),
+    ];
+
+    const onEvent: OnEvent = async (type, data, device) => {
+        await readAttributeIfOnline(type, data, device, attribute);
```
Comment (Koenkk) [inline src/devices/sunricher.ts:160]: I think these only  need to be read one time after the device has joined? If yes, put them in a `configure`.

File: src/devices/sunricher.ts [hunk 2]
```diff
@@ -25,13 +25,150 @@ import {
...
+                    };
+                }
+                return undefined;
+            },
+        },
+    ];
+
+    const toZigbee: Tz.Converter[] = [
+        {
+            key: ['minimumPWM'],
```
Comment (Koenkk) [inline src/devices/sunricher.ts:134]:
```suggestion
            key: ['minimum_pwm'],
```

File: src/devices/sunricher.ts [hunk 3]
```diff
@@ -25,13 +25,150 @@ import {
...
+                await entity.write('genBasic', {[attribute]: {value: numericValue, type: dataType}}, {manufacturerCode: sunricherManufacturerCode});
+                return {state: {externalSwitchType: value}};
+            },
+            convertGet: async (entity, key, meta) => {
+                await entity.read('genBasic', [attribute], {manufacturerCode: sunricherManufacturerCode});
+            },
+        } satisfies Tz.Converter,
+    ];
+
+    const exposes: Expose[] = [e.enum('externalSwitchType', ea.ALL, ['Push button', 'Normal on/off', 'Three way']).withLabel('External Switch Type')];
```
Comment (Koenkk) [inline src/devices/sunricher.ts:98]:
```suggestion
    const exposes: Expose[] = [e.enum('external_switch_type', ea.ALL, ['push_button', 'normal_on_off', 'three_way']).withLabel('External Switch Type')];
```

**Comment by maginawin** [comment]
Hi @Koenkk , thank you for your detailed suggestions, I've modified them.

**Comment by Gnarfoz** [comment]
@maginawin Hi, this change seems aimed at SR-ZG9040A/ZG9041A-D but it also affects https://www.zigbee2mqtt.io/devices/ZG9101SAC-HP-Switch.html.
The external switch attribute shows up, but I don't think that's configurable on these devices?
The PWM thing doesn't show up. The switch is not a dimmer, anyway.

**Pull request merged**